BacktraceScreen.c

```diff
@@ -0,0 +1,384 @@
+   "F2",
+#endif
+   "F3",
+   "Esc",
+   NULL
+};
+
+static const int BacktraceScreenEvents[] = {
+   KEY_F(1),
+#if defined(HAVE_DEMANGLING)
+   KEY_F(2),
+#endif
+   KEY_F(3),
+   27,
+};
+
+static void BacktraceFrame_displayHeader(const BacktraceFrame* frame, RichString* out) {
+   const BacktracePanelPrintingHelper* printingHelper = &frame->backtracePanel->printingHelper;
+   assert(printingHelper);
+
+   const BacktraceScreenDisplayOptions displayOptions = frame->backtracePanel->displayOptions;
+
+   int maxFunctionNameLength = printingHelper->maxFuncNameLen;
+   if (!!(displayOptions & DEMANGLE_NAME_FUNCTION) &&
+         printingHelper->maxDemangledFuncNameLen > 0) {
+      maxFunctionNameLength = printingHelper->maxDemangledFuncNameLen;
+   }
+
+   char* line = NULL;
+   int len = xAsprintf(&line, "%-*s %-*s %-*s %-*s",
+      (int)printingHelper->maxFrameNumLen, BACTRACE_PANEL_HEADER_NUMBER_FRAME,
+      (int)printingHelper->maxAddrLen, BACTRACE_PANEL_HEADER_ADDRESS,
+      (int)maxFunctionNameLength, BACTRACE_PANEL_HEADER_NAME,
+      (int)printingHelper->maxObjPathLen, BACTRACE_PANEL_HEADER_PATH
+   );
+
+   RichString_appendnAscii(out, CRT_colors[BACKTRACE_HEADER], line, len);
+   free(line);
+}
+
+static void BacktraceFrame_highlightBasename(const BacktraceFrame* frame, RichString* out, char* line, int objectPathStart) {
+   assert(objectPathStart >= 0);
+
+   char* basename = frame->backtracePanel->process->procExe + frame->backtracePanel->process->procExeBasenameOffset;
+   char* object = line + objectPathStart;
+   if (!basename || !object) {
+      return;
+   }
+
+   char* lastSlash = strrchr(object, '/');
+   if (lastSlash) {
+      object = lastSlash + 1;
+   }
+
+   char* objectTrimmed = String_trim(object);
+   if (String_eq(basename, objectTrimmed)) {
+      RichString_setAttrn(out, CRT_colors[PROCESS_BASENAME], object - line, strlen(objectTrimmed));
+   }
+
+   free(objectTrimmed);
+}
+
+static void BacktraceFrame_display(const Object* super, RichString* out) {
+   const BacktraceFrame* frame = (const BacktraceFrame* )super;
+   assert(frame);
+
+   if (frame->index == BACKTRACE_FRAME_HEADER_INDEX) {
+      BacktraceFrame_displayHeader(frame, out);
+      return;
+   }
+
+   const BacktracePanelPrintingHelper* printingHelper = &frame->backtracePanel->printingHelper;
+   assert(printingHelper);
+
+   const BacktraceScreenDisplayOptions displayOptions = frame->backtracePanel->displayOptions;
+
+   char* functionName = frame->functionName;
+   int maxFunctionNameLength = printingHelper->maxFuncNameLen;
+   if (!!(displayOptions & DEMANGLE_NAME_FUNCTION) &&
+         printingHelper->maxDemangledFuncNameLen > 0) {
+      maxFunctionNameLength = printingHelper->maxDemangledFuncNameLen;
+      if (frame->demangleFunctionName) {
+         functionName = frame->demangleFunctionName;
+      }
+   }
+
+   char *completeFunctionName = NULL;
+   xAsprintf(&completeFunctionName, "%s+0x%lx", functionName, frame->offset);
```

Comment:
```suggestion
   xAsprintf(&completeFunctionName, "%s+0x%zx", functionName, frame->offset);
```
Don't assume `long` and `size_t` have the same size.

Reply:
Okay, I changed it! Thanks.